proxy/proxy.go

```diff
@@ -468,10 +468,18 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 	}
 }
 
-func setRequestURLForLoadBalancedBackend(u *url.URL, rt *routing.Route, lbctx *routing.LBContext) *routing.LBEndpoint {
+func selectEndpoint(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
+	rt := ctx.route
```

Comment:
minor: I don't understand why do we need rt variable here and why cannot we inline `ctx.route`?

Reply:
rt.Apply